addons/auth_signup/controllers/main.py

```diff
@@ -137,7 +137,7 @@ def do_signup(self, qcontext):
     def _signup_with_values(self, token, values):
         db, login, password = request.env['res.users'].sudo().signup(values, token)
         request.env.cr.commit()     # as authenticate will use its own cursor we need to commit the current transaction
-        uid = request.session.authenticate(db, login, password)
+        uid = request.session.authenticate(request.db, login, password)
```

Comment:
in which case db will be different from request.db ?

Reply:
When you create a new database via the db manager, you want to create a logged-in session in the new database so that the user creating the database doesn't lands on /web in the new db instead of /web/login.

Here is the code, `web/controllers/main.py` L1097

```py
    @http.route('/web/database/create', type='http', auth="none", methods=['POST'], csrf=False)
    def create(self, master_pwd, name, lang, password, **post):
        insecure = odoo.tools.config.verify_admin_password('admin')
        if insecure and master_pwd:
            dispatch_rpc('db', 'change_admin_password', ["admin", master_pwd])
        try:
            if not re.match(DBNAME_PATTERN, name):
                raise Exception(_('Invalid database name. Only alphanumerical characters, underscore, hyphen and dot are allowed.'))
            # country code could be = "False" which is actually True in python
            country_code = post.get('country_code') or False
            dispatch_rpc('db', 'create_database', [master_pwd, name, bool(post.get('demo')), lang, password, post['login'], country_code, post['phone']])
            request.session.authenticate(name, post['login'], password)
            return request.redirect('/web')
        except Exception as e:
            error = "Database creation error: %s" % (str(e) or repr(e))
        return self._render_template(error=error)
```

Reply:
But in this very case, I should use `db` from L138 instead of `request.db`. Fixing it...